src/compiler/core.ts

```diff
@@ -1312,4 +1333,49 @@ namespace ts {
             : ((fileName) => fileName.toLowerCase());
     }
 
+    /**
+     * This isn't the strictest deep equal, but it's good enough for us
+     *  - +0 === -0 (though who really wants to consider them different?)
+     *  - arguments and arrays can be equal (both typeof === object, both have enumerable keys)
+     *  - doesn't inspect es6 iterables (not that they're used in this code base)
+     *  - doesn't inspect regex toString value (so only references to the same regex are equal)
+     *  - doesn't inspect date primitive number value (so only references to the same date are equal)
+     */
+    export function deepEqual(a: any, b: any, memo?: [any, any][]): boolean {

```

Comment:
Extension arguments objects can be literally any object, and that's what's
getting compared. I think the only thing this handles which I shouldn't
need to worry about for that is circularly referential objects. (Which is
why all the shortcomings listed in the comment were OK)

On Fri, Jul 8, 2016, 4:22 PM Nathan Shively-Sanders <
notifications@github.com> wrote:

> In src/compiler/core.ts
> <https://github.com/Microsoft/TypeScript/pull/9038#discussion_r70154465>:
>
> > @@ -1312,4 +1333,49 @@ namespace ts {
> >              : ((fileName) => fileName.toLowerCase());
> >      }
> >
> > +    /**
> > +     * This isn't the strictest deep equal, but it's good enough for us
> > +     *  - +0 === -0 (though who really wants to consider them different?)
> > +     *  - arguments and arrays can be equal (both typeof === object, both have enumerable keys)
> > +     *  - doesn't inspect es6 iterables (not that they're used in this code base)
> > +     *  - doesn't inspect regex toString value (so only references to the same regex are equal)
> > +     *  - doesn't inspect date primitive number value (so only references to the same date are equal)
> > +     */
> > +    export function deepEqual(a: any, b: any, memo?: [any, any][]): boolean {
>
> do we really need all the features that deepEqual provides for its single
> use? I feel like it would be safer to put a utility in services that does
> only the equality that is needed there, and only generalise it as we find
> new uses for it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Microsoft/TypeScript/pull/9038/files/ee23587b1ac09aa5ac4f0c87c6bb8af25839d04c#r70154465>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACzAMroSTC2lG2Ak_PGLEtp796xpbIoyks5qTtvPgaJpZM4IxgwE>
> .
>